src/prefect/agent/docker/agent.py

```diff
@@ -419,11 +443,18 @@ def deploy_flow(self, flow_run: GraphQLResult) -> str:
             networking_config=networking_config,
             labels=labels,
         )
```

Comment:
Small note: During debugging this method was giving some difficult to interpret `Error while deploying flow: APIError(HTTPError('400 Client Error: Bad Request for url: http+docker://localhost/v1.41/containers/create'))` errors. The logging seems to remove the stack-trace.

To make debugging easier I surrounded this with a try-except block that would log the exception with stack-trace and re-raise it. Is this something we would want to put in by default to make logging of exceptions coming from the Docker API more verbose?

Reply:
What information does the original stack track contain that helped you diagnose? Do you have an example you could post?

Reply:
Printing the original stacktrace with `self.logger.exception` prints the following for example:
```
400 Client Error for http+docker://localhost/v1.41/containers/create: Bad Request ("Container cannot be connected to network endpoints: prefect-server, kafka-network")
Traceback (most recent call last):
  File "/home/peter/miniconda3/envs/prefect/lib/python3.8/site-packages/docker/api/client.py", line 268, in _raise_for_status
    response.raise_for_status()
  File "/home/peter/miniconda3/envs/prefect/lib/python3.8/site-packages/requests/models.py", line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http+docker://localhost/v1.41/containers/create

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/peter/workspace/prefect/src/prefect/agent/docker/agent.py", line 433, in deploy_flow
    container = self.docker_client.create_container(
  File "/home/peter/miniconda3/envs/prefect/lib/python3.8/site-packages/docker/api/container.py", line 430, in create_container
    return self.create_container_from_config(config, name)
  File "/home/peter/miniconda3/envs/prefect/lib/python3.8/site-packages/docker/api/container.py", line 441, in create_container_from_config
    return self._result(res, True)
  File "/home/peter/miniconda3/envs/prefect/lib/python3.8/site-packages/docker/api/client.py", line 274, in _result
    self._raise_for_status(response)
  File "/home/peter/miniconda3/envs/prefect/lib/python3.8/site-packages/docker/api/client.py", line 270, in _raise_for_status
    raise create_api_error_from_http_exception(e)
  File "/home/peter/miniconda3/envs/prefect/lib/python3.8/site-packages/docker/errors.py", line 31, in create_api_error_from_http_exception
    raise cls(e, response=response, explanation=explanation)
docker.errors.APIError: 400 Client Error for http+docker://localhost/v1.41/containers/create: Bad Request ("Container cannot be connected to network endpoints: prefect-server, kafka-network")
```

Reply:
Yeah let's expose this stack trace it has much more detail!

Reply:
Modified the code to expose the stacktrace